### ISSUE TITLE
Hide blank section of git when not in a git directory

### ISSUE THREAD

**Issue by jeremyFreeAgent** [jeremyFreeAgent/oh-my-zsh-powerline-theme#16]

**Comment by protist** [comment]
This would be great. Probably 99% of my time I am not in a git directory, and there are two extraneous characters in the prompt. (There are also four unnecessary characters at the beginning of the prompt with username and hostname hidden, if you don't care about the green/red colours, and 3–4 extra on the right prompt with no error and `POWERLINE_RIGHT_A="exit-status-on-fail"`.)

**Comment by jeremyFreeAgent** [comment]
Hi @protist thanks for your feedback. I'm currently doing some improvements, I will look at that.

**Comment by protist** [comment]
Thank you. FWIW out of all the powerline-inspired zsh prompts, I already do find yours to be the most compact, which I appreciate. Thanks for that.

**Comment by protist** [comment]
While you are looking at improvements, there is one other space-related measure that gets on my OCD. The right prompt is one space short of the right margin! I tested in both `terminator` and `konsole`.

![screenshot](http://i.imgur.com/BCNOxjA.png)

**Comment by protist** [comment]
Sorry, I just saw issue #3.

**Comment by jeremyFreeAgent** [comment]
@protist Yes this is a well known issue. It's not due to the theme but the terminal. I can't find an correct fix.

**Comment by jeremyFreeAgent** [comment]
@protist 
> Thank you. FWIW out of all the powerline-inspired zsh prompts, I already do find yours to be the most compact, which I appreciate. Thanks for that.

Thanks, I wanted to be simple without email of weather bullshits. Event if there are a lot of options now it's only about custom design not feature.
